### DISCUSSION TITLE
Timer jumps back after set to 0

### DISCUSSION THREAD

**Discussion by DennisWarchocki** [neuefische/questions#271] [Web] [answered]
Hello everyone,

I have a Problem with my Timer, I made a reset button to reset my previous time input and it looks like this:
`					
						<button
							onClick={event => {
								event.preventDefault;
								setTimerHours(0);
								setTimerMinutes(0);
								setTimerSeconds(0);
							}}
						>`

<img width="281" alt="Bildschirmfoto 2022-10-16 um 15 51 06" src="https://user-images.githubusercontent.com/109968361/196039116-5ec2bba6-2156-4cbe-92ec-3a978ddcdd68.png">

If Im pressing the reset button now, The time goes down to 0 for like 1 Second, and then its returning to the previous value.
I couldnt figure out what creates this behaviour, any ideas?

preview:
https://capstone-project-git-styling-denniswarchocki.vercel.app/timer

link to my timer code:
https://github.com/DennisWarchocki/capstone-project/blob/main/src/pages/timer.js

**Comment by DennisWarchocki** [comment]
I used a refresh function now to solve the Issue:
```

	function refresh() {
		window.location.reload(false);
	}
```

and the button looks like this now:
`							<button
								onClick={event => {
									event.preventDefault;
									refresh();
								}}
							>
								Reset
							</button>`

But I think that there is a smarter way to solve the Problem.

**Comment by mbosselmann** [comment, marked as answer]
Hi Dennis,
I think the functionality of your reset button needs to be updated to work correct. Add `type="reset"` and  declare that `userInputHours` and `userInputMinutes` should be updated with `0` on button click:

```js
<button
	onClick={() => {
		setUserInputHours(0);
		setUserInputMinutes(0);
	}}
	type="reset"
	>
	Reset
</button>
``` 

I hope my suggestion does help. :)

**Reply by DennisWarchocki** [reply]
Thank you Mareike :)
